### PR TITLE
Remove ISO 8601 date.parse shim code

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: packages/ember-data/lib/ext/date.js [hunk 1]
```diff
@@ -16,46 +16,13 @@
...
-  } else {
-    timestamp = origParse ? origParse(date) : NaN;
-  }
-
-  return timestamp;
-};
+Ember.Date.parse = origParse;
 
 if (Ember.EXTEND_PROTOTYPES === true || Ember.EXTEND_PROTOTYPES.Date) {
   Date.parse = Ember.Date.parse;
```
Comment (pangratz) [inline packages/ember-data/lib/ext/date.js]: I guess this line doesn't make sense then anymore, right?

Comment (bmac) [inline packages/ember-data/lib/ext/date.js]: Good call.

Comment (pangratz) [inline packages/ember-data/lib/ext/date.js]: Save all the bytes! :hammer:

**Comment by bmac** [comment]
We have decided to keep this code in the date transform and warn the user if there is no timezone or the timezone is not specified.

**Pull request closed without merging**